src/utils/ethereum.js

```diff
@@ -34,127 +34,141 @@ export const FEE_UI_UNSCALED = decimalUnscaling(
   SCALING_DECIMALS
 );
 
-export const getWeb3 = () =>
-  new Promise(async (resolve, reject) => {
-    
+let isWindowActive = true; // Flag to track window visibility
+
+window.addEventListener("visibilitychange", () => {
+  isWindowActive = document.visibilityState === "visible";
+  console.log("Tab visibility changed:", isWindowActive ? "Active" : "Inactive");
+});
+
+const withWindowVisibilityCheck = (action) => {
+  return (...args) => {
+    if (!isWindowActive) {
+      throw new Error("Window is not active. Please activate the tab.");
+    }
+    return action(...args);
+  };
+};
+
+export const getWeb3 = withWindowVisibilityCheck(
+  () =>
+    new Promise(async (resolve, reject) => {
       try {
         const web3 = new Web3(BLOCKCHAIN_URI);
         resolve(web3);
       } catch (error) {
         reject(error);
       }
+    })
+);
 
-  });
-
-export const getDjedContract = (web3) => {
+export const getDjedContract = withWindowVisibilityCheck((web3) => {
   const djed = new web3.eth.Contract(djedArtifact.abi, DJED_ADDRESS);
   return djed;
-};
+});
 
-export const getOracleAddress = async (djedContract) => {
+export const getOracleAddress = withWindowVisibilityCheck(async (djedContract) => {
   return await web3Promise(djedContract, "oracle");
-};
-
-export const getOracleContract = (web3, oracleAddress, msgSender) => {
-  const oracle = new web3.eth.Contract(oracleArtifact.abi, oracleAddress, {
-    from: msgSender
-  });
-  return oracle;
-};
+});
+
+export const getOracleContract = withWindowVisibilityCheck(
+  (web3, oracleAddress, msgSender) => {
+    const oracle = new web3.eth.Contract(oracleArtifact.abi, oracleAddress, {
+      from: msgSender
+    });
+    return oracle;
+  }
+);
 
-export const getCoinContracts = async (djedContract, web3) => {
+export const getCoinContracts = withWindowVisibilityCheck(async (djedContract, web3) => {
```

Comment:
Is this query being done repeatedly?

If not (as I would expect), we shouldn't need to check window visisbility.

We should only check window visibility for queries that are being made repeatedly. If the frontend has been well-designed (and it may not have been), I think only the queries that fetch prices should be made repeatedly.

We need to find the place in the code that is causing the repetition. Could you find it and show it to me?

Reply:
@ceilican When I manually inspected the functions again, I discovered that two of them, `getCoinDetails` and `getAccountDetails`, continue to fetch prices even when the user is inactive on the window. Specifically,     `getCoinDetails` makes 7 requests per minute, while `getAccountDetails` makes 3 requests per minute. 
Together, they total 10 requests per minute.